### PR TITLE
scope environment variable groups by project and not by stage

### PR DESCRIPTION
**Pull request by grosser**
1 project has a set of settings, stages are just a selection of deploy groups

@zendesk/runway 

### Risks
 - None

### CODE REVIEW

File: plugins/env/test/controllers/environment_variable_groups_controller_test.rb [hunk 1]
```diff
@@ -2,8 +2,9 @@
 
 describe Admin::EnvironmentVariableGroupsController do
   let(:stage) { stages(:test_staging) }
+  let(:project) { stage.project }
   let(:deploy_group) { stage.deploy_groups.first }
-  let(:env_group) do
+  let!(:env_group) do
```
Comment (sandlerr) [inline plugins/env/test/controllers/environment_variable_groups_controller_test.rb]: let! is making a test fail, there is G1 on line 63 but uniqueness constraint

Comment (grosser) [inline plugins/env/test/controllers/environment_variable_groups_controller_test.rb]: fixed, good to go ?

File: plugins/env/app/controllers/admin/environment_variable_groups_controller.rb [hunk 2]
```diff
@@ -34,10 +34,10 @@ def destroy
     def preview
       @groups = if params[:group_id]
         @group = EnvironmentVariableGroup.find(params[:group_id])
-        SamsonEnv.env_groups(Stage.new(environment_variable_groups: [@group], deploy_groups: DeployGroup.all))
+        SamsonEnv.env_groups(Stage.new(project: Project.new(environment_variable_groups: [@group]), deploy_groups: DeployGroup.all))
```
Comment (sandlerr) [inline plugins/env/app/controllers/admin/environment_variable_groups_controller.rb]: This signature is becoming kinda complicated, If you were starting from scratch would it still take a stage? Maybe a project and an array or splat of deploy groups?

Comment (sandlerr) [inline plugins/env/app/controllers/admin/environment_variable_groups_controller.rb]: I guess this applies to other methods around this plugin as well so no prob if you want to leave it as is

**Comment by sandlerr** [comment]
:+1:

**Comment by grosser** [comment]
updated the interface to project + deploy_groups

**Pull request merged**